src/components/Transactions/transactions.test.tsx

```diff
@@ -41,7 +41,7 @@ describe('Transactions table component', () => {
       </BrowserRouter>
     );
 
-    const button = screen.getByRole('button');
+    const button = screen.getAllByRole('button')[0];
```

Comment:
This way of catch elements is breakable cause a minimum change of DOM structure could fail the function.
Take a look to https://testing-library.com/docs/queries/bytestid

Reply:
Actually, eliminating the button solved the problem and I can revert to "getByRole", should I still implement the test with the test id?

Reply:
Yes, testing using an id is definitely better